panther_hardware_interfaces/include/panther_hardware_interfaces/gpio_driver.hpp

```diff
@@ -1,33 +1,101 @@
 #ifndef PANTHER_HARDWARE_INTERFACES__GPIO_CONTROLLER_HPP_
 #define PANTHER_HARDWARE_INTERFACES__GPIO_CONTROLLER_HPP_
 
+#include <atomic>
 #include <thread>
+#include <vector>
+#include <poll.h>
 
+#include <boost/function.hpp>
 #include <gpiod.hpp>
 
 namespace panther_hardware_interfaces
 {
+enum class GPIOpins {
+  WATCHDOG = 0,
+  E_STOP_RESET,
+  MOTOR_DRIVER_EN,
+  VMOT_ON,
+  FAN_SW,
+  LED,
+  TEST_IN,
+  UNKNOWN
+};
+
+GPIOpins operator+(const GPIOpins & pin, int int_val);
 
-// Placeholder GPIO class used for simple enabling, will be replaced by proper implementation in the future
+struct GPIOinfo
+{
+  std::string name;
+  gpiod::line::direction direction;
+  gpiod::line::bias bias = gpiod::line::bias::AS_IS;
+  gpiod::line::value init_value = gpiod::line::value::INACTIVE;
+  gpiod::line::value value = gpiod::line::value::INACTIVE;
+  gpiod::line::offset offset = -1;
+};
 
-enum class GPIOPins { VMOT_ON = 6, MOTOR_DRIVER_EN = 23, WATCHDOG = 14, E_STOP_RESET = 27 };
+struct OperationResult
+{
+  bool success;
+  std::string message = "";
+};
 
 class GPIOController
```

Comment:
I think that it may make sense to split it into two classes - one more generic and lower level (set_pin_value, get_pin_value, etc) and one more specific for panther with watchdog thread, etc.